greedy_methods/fractional_cover_problem.py

```diff
@@ -0,0 +1,108 @@
+# Explanation:- https://en.wikipedia.org/wiki/Set_cover_problem
+
+from dataclasses import dataclass
+
+
+@dataclass
+class Item:
+    weight: int
+    value: int
+
+    def __eq__(self, other: object) -> bool:
```

Comment:
One of the really cool features of dataclasses is that this comes for free.  If you rename this function to `not_really__eq__()`, all your doctests still pass!!